plugins/hypervisors/kvm/src/main/java/com/cloud/hypervisor/kvm/storage/LibvirtStorageAdaptor.java

```diff
@@ -775,23 +781,22 @@ public KVMPhysicalDisk createPhysicalDisk(String name, KVMStoragePool pool,
         s_logger.info("Attempting to create volume " + name + " (" + pool.getType().toString() + ") in pool "
                 + pool.getUuid() + " with size " + toHumanReadableSize(size));
 
-        switch (pool.getType()) {
-            case RBD:
-                return createPhysicalDiskByLibVirt(name, pool, PhysicalDiskFormat.RAW, provisioningType, size);
-            case NetworkFilesystem:
-            case Filesystem:
-                switch (format) {
-                    case QCOW2:
-                    case RAW:
-                        return createPhysicalDiskByQemuImg(name, pool, format, provisioningType, size, passphrase);
-                    case DIR:
-                    case TAR:
-                        return createPhysicalDiskByLibVirt(name, pool, format, provisioningType, size);
-                    default:
-                        throw new CloudRuntimeException("Unexpected disk format is specified.");
-                }
-            default:
-                return createPhysicalDiskByLibVirt(name, pool, format, provisioningType, size);
+        StoragePoolType poolType = pool.getType();
```

Comment:
can we use `switch (poolType /*.toString*/)` here? jabva supports switching on strings, so no reason to do this refactor.

Reply:
tried switch/case on poolType (string), noticed some errors with it. keeping the new/refactored changes.